numpy/testing/_private/utils.py

```diff
@@ -706,6 +711,20 @@ def chk_same_position(x_id, y_id, hasval='nan'):
                                 % (hasval), verbose=verbose, header=header,
                                 names=('x', 'y'), precision=precision)
             raise AssertionError(msg)
+        return x_id if x_id.shape else y_id
```

Comment:
I think this is backward - you should return the scalar boolean mask here, not the array one (which would at this point be equal to the larger array anyway). That allows you to remove all of your special cases in `remove_flagged`

Reply:
Yes, indeed, you are completely right - that gets rid of `remove_flagged` altogether. Now implemented.